src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -641,15 +644,63 @@ int goto_instrument_parse_optionst::doit()
 
     if(
       cmdline.isset("dump-c") || cmdline.isset("dump-cpp") ||
-      cmdline.isset("dump-c-type-header"))
+      cmdline.isset("contract-to-stub") || cmdline.isset("dump-c-type-header"))
     {
       const bool is_cpp=cmdline.isset("dump-cpp");
+      const bool c_stub = cmdline.isset("contract-to-stub");
       const bool is_header = cmdline.isset("dump-c-type-header");
       const bool h_libc=!cmdline.isset("no-system-headers");
       const bool h_all=cmdline.isset("use-all-headers");
       const bool harness=cmdline.isset("harness");
       namespacet ns(goto_model.symbol_table);
 
+      if(c_stub)
+      {
+        bool okay = true;
+        std::stringstream ss;
+        ss << "--contract-to-stub cannot be used together with ";
+        for(const char *flag : {"dump-c",
+                                "dump-cpp",
+                                "dump-c-type-header",
+                                "no-system-headers",
+                                "use-all-headers",
+                                "harness"})
+        {
+          if(cmdline.isset(flag))
+          {
+            if(!okay)
+              ss << "or ";
+            ss << "--" << flag << " ";
+            okay = false;
+          }
+        }
+
+        if(!okay)
+        {
+          std::string correct_input_string = ss.str();
+          throw invalid_command_line_argument_exceptiont(
+            "Invalid flag combination",
+            "--contract-to-stub",
+            correct_input_string);
+        }
+
+        // Modify the goto_model to only contain the body of the stub
+        // and the necessary declarations
+        std::string function_name = cmdline.get_value("contract-to-stub");
+        log.status() << "Creating stub from contract for: " << function_name
+                     << messaget::eom;
+        if(function_stubs(goto_model, function_name, log))
```

Comment:
if `--contract-to-stub` should always mean dumping just the stubbed function then do the `goto_model.clear()` etc here, not in `function_stubs`

Reply:
Thanks, I implemented that :)